sdk/avalon_sdk/work_order/work_order_params.py

```diff
@@ -115,37 +124,48 @@ def set_session_key_iv(self, session_iv):
+            raise TypeError("requesterNonce cannot be None")
         self.params_obj["requesterNonce"] = requester_nonce
 
     def add_encrypted_request_hash(self):
         """
         Calculates request hash based on EEA trusted-computing spec 6.1.8.1
         and set encryptedRequestHash parameter in the request.
         """
-        sig_obj = signature.ClientSignature()
-        concat_string = self.get_requester_nonce() + \
-            self.get_work_order_id() + \
-            self.get_worker_id() + \
-            self.get_workload_id() + \
-            self.get_requester_id()
-        concat_bytes = bytes(concat_string, "UTF-8")
-        # SHA-256 hashing is used
-        hash_1 = crypto_utility.byte_array_to_base64(
-            crypto_utility.compute_message_hash(concat_bytes)
-        )
-        hash_2 = sig_obj.calculate_datahash(self.get_in_data())
-        hash_3 = ""
-        out_data = self.get_out_data()
-        if out_data and len(out_data) > 0:
-            hash_3 = sig_obj.calculate_datahash(out_data)
-        concat_hash = hash_1 + hash_2 + hash_3
-        concat_hash = bytes(concat_hash, "UTF-8")
-        self.final_hash = crypto_utility.compute_message_hash(concat_hash)
-        encrypted_request_hash = crypto_utility.encrypt_data(
-            self.final_hash, self.session_key, self.session_iv)
-        encrypted_request_hash_hex = crypto_utility.byte_array_to_hex(
-                                        encrypted_request_hash)
-        self.params_obj["encryptedRequestHash"] = encrypted_request_hash_hex
+        try:
+            sig_obj = signature.ClientSignature()
+            concat_string = self.get_requester_nonce() + \
+                self.get_work_order_id() + \
+                self.get_worker_id() + \
+                self.get_workload_id() + \
+                self.get_requester_id()
+            concat_bytes = bytes(concat_string, "UTF-8")
+            # SHA-256 hashing is used
+            hash_1 = crypto_utility.byte_array_to_base64(
+                crypto_utility.compute_message_hash(concat_bytes)
+            )
+            hash_2 = sig_obj.calculate_datahash(self.get_in_data())
+            hash_3 = ""
+            out_data = self.get_out_data()
+            if out_data and len(out_data) > 0:
+                hash_3 = sig_obj.calculate_datahash(out_data)
+            concat_hash = hash_1 + hash_2 + hash_3
+            concat_hash = bytes(concat_hash, "UTF-8")
+            self.final_hash = crypto_utility.compute_message_hash(concat_hash)
+            encrypted_request_hash = crypto_utility.encrypt_data(
+                self.final_hash, self.session_key, self.session_iv)
+            encrypted_request_hash_hex = crypto_utility.byte_array_to_hex(
+                                            encrypted_request_hash)
+            self.params_obj["encryptedRequestHash"] = \
+                encrypted_request_hash_hex
+            return True, ""
+
+        except Exception as err:
```

Comment:
Did we really want to pass a tuple here? As the second value i.e.- the error response is always fixed. returning True/False would have sufficed.